### DISCUSSION TITLE
Is it possible to use yup validation from within the input components themselves?

### DISCUSSION THREAD

**Discussion by Eugleo** [react-hook-form/react-hook-form#3907] [Q&A]
Hey, I'm making a form-building "DSL" for some of my less technically minded colleagues. The DSL uses jsx and looks like this:

```jsx
<Question>What type of the analysis do you want to perform?</Question>
<SingleChoice id="analysisType">
  <Option value="Purified proteins">
    <Question>How did you purify them?</Question>
    <ShortText id="purificationMethod" />
    <Question>No, really, tell us in detail!</Question>
    <LongText id="purificationMethodDetail" />
  </Option>
  <Option value="Cell lysate"/>
  <Option value="Tissue lysate"/>
</SingleChoice>
```

I'd like for them to be able to set a field as required simply by doing `<LongText id="..." required />`, but at the same time I'd like to use `yup` for the validation. Is that possible, to tell yup "I'm required" from within the component? Normally I'd have to build a schema object on the top level and pass it to the `form` component, which would be hard for my colleagues to do.

**Comment by bluebill1049** [comment]
you can't use build-in validator with schema/resolvers.
